### PR TITLE
Add basic settings

### LINKED ISSUE #2: Add basic settings
These settings should be associated with a section/page belonging to the tool_messagebroker plugin.
#### Message data-store selector
Drop down to choose which data store to use among those installed in messagebrokerdatastore_* (can be forced through $CFG, defaults to standarddb) 
#### Message Receiving Mode selector
Drop down to choose Durable, Ephemeral or Receiver Preference mode with Receiver Preference mode as the default.
This will define how messages are handled when they come into the message broker. Either all Receivers will be treated as Ephemeral, Durable or be allowed to express their preference.
#### Durable Messages per Task Execution
Counter that defines up to how many durable messages should be processed per execution of the Durable Message Processor Task: range 1 - INTMAX

### PR DESCRIPTION
**Pull request by cameron1729**
# Testing
1. Browse to "Site administration" > "Plugins" > "Admin tools" > "Message Broker"
2. **Verify** settings display correctly 

Fixes #2

### CODE REVIEW

File: lang/en/tool_messagebroker.php [hunk 1]
```diff
@@ -12,5 +12,14 @@
...
+$string['datastore_help'] = 'The selected datastore plugin will be used for persisting data associated with messages.';
+$string['messagespertask'] = 'Messages per task';
+$string['messagespertask_help'] = 'The maximum number of messages which will be processed per execution of the Durable Message Processor task.';
 $string['pluginname'] = 'Message Broker';
 $string['privacy:metadata'] = 'The Message Broker plugin does not store any personal data';
+$string['processingstyle:durable'] = 'Durable';
+$string['processingstyle:ephemeral'] = 'Ephemeral';
+$string['processingstyle:receiver_preference'] = 'Receiver preference';
+$string['receivemode'] = 'Message receiving mode';
+$string['receivemode_help'] = 'Specifies how receivers will be treated when a message enters the system. When a value is selected, all receivers will be treated as if they interested in that type of message. When set to "Receiver preference" are given the option to ignore messages they have not expressed preference for.';
```
Comment (darrencocco) [inline lang/en/tool_messagebroker.php]: Not quite right.

`Specifies how receivers will be treated when a message enters the system. When a value is selected, all receivers will be sent messages through the selected process (ephemeral/durable). When set to "Receiver preference" then the receivers preferred method will be used.

- Ephemeral: Receiver will receive and be required to process the message the before the message is confirmed as received by Moodle. (Moodle doesn't retry these receivers)
- Durable: Receiver will receive and process the message in the background. (Moodle will continue to retry the message internally until the receiver gives a positive response)`

**Comment by darrencocco** [comment]
Looks good other than the string change requested.

**Comment by cameron1729** [comment]
@darrencocco - sorry, not quite sure what string change that is?

**Comment by darrencocco** [comment]
the receivemode_help string

**Comment by darrencocco** [comment]
wat

Not quite right.

'Specifies how receivers will be treated when a message enters the system. When a value is selected, all receivers will be sent messages through the selected process (ephemeral/durable). When set to "Receiver preference" then the receivers preferred method will be used.

 - Ephemeral: Receiver will receive and be required to process the message the before the message is confirmed as received by Moodle. (Moodle doesn't retry these receivers)
 - Durable: Receiver will receive and process the message in the background. (Moodle will continue to retry the message internally until the receiver gives a positive response)'

File: lang/en/tool_messagebroker.php (hunk 1 above)
Comment (cameron1729) [reply lang/en/tool_messagebroker.php]: Pushed a commit to update the help string.

**Pull request merged**